src/math_tools.rs

```diff
@@ -46,10 +85,69 @@ mod tests {
 
     use super::*;
 
-    #[test]
-    fn test_shannon_entropy() {
-        let input = b"000111"; // 50% entropy distribution ~ 1 bit information
+    mod shannon {
+        use super::*;
+
+        #[test]
+        fn test_shannon_entropy() {
+            let input = b"000111"; // 50% entropy distribution ~ 1 bit information
+
+            assert_relative_eq!(shannon_entropy(input), 1.0);
+        }
+    }
+
+    mod chi_square {
+        use super::*;
+        use rand_core::{OsRng, RngCore};
 
-        assert_relative_eq!(shannon_entropy(input), 1.0);
+        #[test]
+        fn test_non_uniform_distribution() {
+            let uniform_distribution = [0u8; 4096];
+            let chi_square_value = chi_square_probability(&uniform_distribution);
+
+            // Expect chi-square to be relatively low for uniform random data
```

Comment:
I feel some discrepancy between these comments and the test content themselves. The adjective `uniform` is also somewhat ambiguous. E.g. this case is not a uniform distribution of bits, but is `uniform` in the sense it's all zeros.

Reply:
Fixed it by removing the comment as I felt it was superfluous considering we already have the error message in case the test fail.

The wording has been fixed. It's uniform in the statistical sense. Considering 255 possible values, a fully uniform distribution will hold each value once, and fully non-uniform distribution will hold one value 255 times.